### PR TITLE
Enabled bfloat16 dtype on CUDA

### PR DESCRIPTION
**Pull request by izdeby**
Enabled basic functionality for bfloat16 dtype on CUDA. 
Tested via unit tests.

### CODE REVIEW

**Review by ezyang — APPROVED** [review]
okey dokey

File: aten/src/ATen/cuda/NumericLimits.cuh [hunk 1]
```diff
@@ -94,6 +94,14 @@ struct numeric_limits<at::Half> {
   static inline __host__ __device__ at::Half upper_bound() { return at::Half(0x7C00, at::Half::from_bits()); }
 };
 
+template <>
+struct numeric_limits<at::BFloat16> {
+  static inline __host__ __device__ at::BFloat16 lowest() { return at::BFloat16(0x0080, at::BFloat16::from_bits()); }
+  static inline __host__ __device__ at::BFloat16 max() { return at::BFloat16(0x7F7F, at::BFloat16::from_bits()); }
+  static inline __host__ __device__ at::BFloat16 lower_bound() { return at::BFloat16(0xFF80, at::BFloat16::from_bits()); }
+  static inline __host__ __device__ at::BFloat16 upper_bound() { return at::BFloat16(0x7F80, at::BFloat16::from_bits()); }
```
[the remark this answers is not in the document]
Comment (ezyang) [reply aten/src/ATen/cuda/NumericLimits.cuh]: I guess I should have checked them XD

**Comment by izdeby** [comment]
@ailzhang, this PR breaks XLA

**Comment by suo** [comment]
@izdeby this PR appears to have broken the windows build. I am reverting.

**Comment by izdeby** [comment]
@suo, sorry about that.

**Pull request closed without merging**